sematic/api/endpoints/tests/test_metrics.py

```diff
@@ -26,3 +40,124 @@ def test_run_created(persisted_run: Run):  # noqa: F811
+
+
+@pytest.mark.parametrize(
+    "url, expected_series",
+    (
+        (
+            "foo",
+            {
+                "metric_name": "foo",
+                "series": [[0.5, []]],
+                "metric_type": "GAUGE",
+                "columns": [],
+            },
+        ),
+        (
+            f"foo?labels={json.dumps(dict(function_path='foo'))}",
+            {
+                "metric_name": "foo",
+                "series": [[1, []]],
+                "metric_type": "GAUGE",
+                "columns": [],
+            },
+        ),
+        (
+            "foo?group_by=function_path",
+            {
+                "metric_name": "foo",
+                "series": [[0, ["bar"]], [1, ["foo"]]],
+                "metric_type": "GAUGE",
+                "columns": ["function_path"],
+            },
+        ),
+        (
+            (
+                f"foo?from_time={int(datetime(2023, 4, 12).timestamp() - 1)}"
+                f"&to_time={int(datetime(2023, 4, 12).timestamp() + 1)}"
+            ),
+            {
+                "metric_name": "foo",
+                "series": [[1, []]],
+                "metric_type": "GAUGE",
+                "columns": [],
+            },
+        ),
+        (
+            "foo?rollup=auto",
+            {
+                "metric_name": "foo",
+                "series": [
+                    [0, [datetime(2023, 4, 11).timestamp()]],
+                    [1, [datetime(2023, 4, 12).timestamp()]],
+                ],
+                "metric_type": "GAUGE",
+                "columns": ["timestamp"],
+            },
+        ),
+        (
+            (
+                f"foo?rollup={24 * 3600}"
+                f"&from_time={int(datetime(2023, 4, 10).timestamp())}"
+                f"&to_time={int(datetime(2023, 4, 13).timestamp())}"
+            ),
+            {
+                "metric_name": "foo",
+                "series": [
+                    [0, [datetime(2023, 4, 11).timestamp()]],
+                    [1, [datetime(2023, 4, 12).timestamp()]],
+                ],
+                "metric_type": "GAUGE",
+                "columns": ["timestamp"],
+            },
+        ),
+    ),
+)
+def test_get_metrics_endpoint(
+    url: str,
+    expected_series,
+    persisted_metric_points: List[MetricPoint],  # noqa: F811
+    test_client: flask.testing.FlaskClient,  # noqa: F811
+):
+    response = test_client.get(f"/api/v1/metrics/{url}")
+
+    payload = response.json
+
+    assert payload["content"] == expected_series  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "url, expected_list",
+    (
+        ("", ["bar", "foo"]),
+        (f"?labels={json.dumps(dict(root_function_path='bat'))}", ["foo"]),
+    ),
+)
+def test_list_metrics_endpoint(
+    url: str,
+    expected_list: List[str],
+    persisted_metric_points: List[MetricPoint],  # noqa: F811
+    test_client: flask.testing.FlaskClient,  # noqa: F811
+):
+    response = test_client.get(f"/api/v1/metrics{url}")
```

Comment:
On line 123 there is a `/`.
```suggestion
    response = test_client.get(f"/api/v1/metrics/{url}")
```

Reply:
Yes because it queries a different endpoint.